docs/pull_request_template.md

```diff
@@ -0,0 +1,38 @@
+### All Submissions:
+
+* [ ] Have you followed the guidelines in our **CONTRIBUTING** document?
+* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
+
+<!-- You can erase any parts of this template not applicable to your Pull Request. -->
+
+
+<!-- These have been added according to the rOpenSci guidelines. -->
+<!-- From: https://www.talater.com/open-source-templates/#/page/99 -->
+
+## Types of changes
+<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
+- [ ] Bug fix (non-breaking change which fixes an issue. List the bug ID if applicable)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing core functionality to change) 
+
+
+### New Feature Submissions checklist:
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+1. [ ] Have you ensured there are no conflicts with major packages (e.g. tidyverse, MASS) or base R packages?
+2. [ ] Does your new feature require documentation, and if so, have you udpated the documentation?
+3. [ ] Have you added tests for your features, and do your features pass all tests?
+4. [ ] Have you linted your code with lintr locally prior to submission?
+5. [ ] Have you styled your code with styler locally prior to submission?
+6. [ ] Have you checked the performance of your code?
```

Comment:
I assume this just means profiling, but we might want tests to explicitly involve comparison with other implementations or alternative algorithms?

Reply:
I think you are asking: do we require actual bench marking, or do we just want to know if the code will slow down the overall package in a major way? Or are we just trying to avoid sloppy code. We should ask for some evidence of the performance, but against what benchmark?

Reply:
Maybe just ask them to offer benchmarks if they think it's relevant to their contribution?

Reply:
We can expand on this to request that they can include [code profiling](http://adv-r.had.co.nz/Profiling.html) if they think it's relevant.

Reply:
Line profiling is not going to tell you how quickly their code works as a function of dataset size. Benchmarking would usually aim to.

Reply:
We can point them to these [packages](https://www.alexejgossmann.com/benchmarking_r/) if they wish to include benchmarking.

Reply:
Not sure how to edit this one, but happy to follow your suggestions.

Reply:
Can we just write:
6. Did you consider the performance of your code? If possible check performance using these [packages](https://www.alexejgossmann.com/benchmarking_r/)

Maybe "possible" is not the right word. Maybe "needed"?

Coming to think if it, I suggest removing this completely. I am not totally happy with the word "performance" it is too broad and could mean more than "benchmarking"

Reply:
@WillemVervoort do you mean remove point 6 completely, or the second part of your suggestion?

I was going to suggest:
Have you [benchmarked](https://www.alexejgossmann.com/benchmarking_r/) the performance of your code if applicable (e.g. with microbenchmark)?

Reply:
I like Jazmin's wording

Reply:
Yes,Thumbs up for Jazmin's wording, more specific
I was thinking removing 6 totally, but your solution is better